### PR TITLE
Whats new page for 3.1 

### PR DESCRIPTION
**Pull request by story645**
## PR Summary
Put all the what's new entries together into something hopefully vaguely resembling cohesion based on the directions [here](https://matplotlib.org/devdocs/devel/release_guide.html#update-and-validate-the-docs). Was a bit confused as documented in #13853 

Also having trouble rendering the toc properly. 
(ETA: I'll do the API changes compilation too if this one was mostly on target)

### CODE REVIEW

File: doc/users/whats_new.rst [hunk 1]
```diff
@@ -10,251 +10,325 @@ revision, see the :ref:`github-stats`.
...
+keeping all commas. Passing a list of strings from within a Python script still works as it used to.
 
-``:math:`` directive renamed to ``:mathmpl:``
----------------------------------------------
 
-The ``:math:`` rst role provided by `matplotlib.sphinxext.mathmpl` has been
-renamed to ``:mathmpl:`` to avoid conflicting with the ``:math:`` role that
-Sphinx 1.8 provides by default.  (``:mathmpl:`` uses Matplotlib to render math
-expressions to images embedded in html, whereas Sphinx uses MathJax.)
+Added support for MacOSX backend for PyPy
```
Comment (jklymak) [inline doc/users/whats_new.rst]: This is actually a bigger deal, and should probably be higher up:

Matplotlib no longer requires framework app build on MacOSX backend
-----------------------------------------------------------------------

Previous versions of matplotlin required a Framework build of python to
work.  The app type was updated to no longer require this, so the MacOSX
backend should work with non-framework python.

**Comment by tacaswell** [comment]
@story645 Added some commits, rebased, and force-pushed.

File: doc/users/whats_new.rst (hunk 1 above)
Comment (tacaswell) [reply doc/users/whats_new.rst]: moved this up.

**Comment by tacaswell** [comment]
Want to tweak this to use `include` and `ifconfig` like the API changes PR did to take this simpler going forward, but very :zzz: so stopping for the night.

**Comment by story645** [comment]
@tacaswell the include ifconfig stuff is I think what I was confused about in #13853

**Comment by tacaswell** [comment]
If I understand `ifconfig` correctly (and used it correctly), then when we tag and build the docs it should automatically exclude the glob tocs.

**Comment by tacaswell** [comment]
backported and directly pushed in b1a6fba

**Pull request merged**